### PR TITLE
✨ Second cut for new component: [amp-access-fewcents] paywall support by Fewcents

### PR DESCRIPTION
**Pull request by mayank-s-dev**
Ref to I2I: #[31854](https://github.com/ampproject/amphtml/issues/31854)

**Description:**
This PR is second cut for the amp-access-fewcents component, you can find the first PR [here](https://github.com/ampproject/amphtml/pull/37096), and this PR includes:
- Authorize API call completion
- Handled the authorize response
- Completing the paywall
- Added design
- Added amp-analytics using segment.

**References:**
We have been following amp-access-laterpay and amp-access-poool for our reference.

- [amp-access](https://github.com/ampproject/amphtml/tree/main/extensions/amp-access)
- AMP code contribution docs: https://github.com/ampproject/amphtml/blob/main/docs/contributing-code.md
- First time setup: https://github.com/ampproject/amphtml/blob/main/docs/getting-started-e2e.md
- Building an extension: https://github.com/ampproject/amphtml/blob/main/docs/building-an-amp-extension.md
- [amp-analytics](https://amp.dev/documentation/guides-and-tutorials/optimize-and-measure/configure-analytics/analytics_basics/?format=websites#use-amp-pixel-or-amp-analytics)
- Segment AMP : https://segment.com/docs/connections/sources/catalog/libraries/mobile/amp/
- [Style and CSS](https://amp.dev/documentation/guides-and-tutorials/develop/style_and_layout/?format=websites)

### CODE REVIEW

**Comment by mayank-s-dev** [comment]
@calebcordry @powerivq 
Hey Guys, I have worked on the second part of the extension and looking forward for your reviews, Let me know if I could be of any help during the process.

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js [hunk 1]
```diff
@@ -93,9 +139,40 @@ export class AmpAccessFewcents {
...
-    dev().fine(TAG, 'Publishers config', this.fewcentsConfig_);
-    return;
+    const accessKey = this.fewcentsConfig_['accessKey'];
+    const articleIdentifier = this.fewcentsConfig_['articleIdentifier'];
+    const category = this.fewcentsConfig_['category'];
+
+    const {hostname} = parseUrlDeprecated();
+
+    return (
+      CONFIG_BASE_PATH +
```
Comment (powerivq) [inline extensions/amp-access-fewcents/0.1/fewcents-impl.js:149]: Is everything properly escaped (especially URL arguments)?

File: examples/amp-access-fewcents.html [hunk 2]
```diff
@@ -1,3 +1,18 @@
+<!--
+  Copyright 2021 The AMP HTML Authors. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (mayank-s-dev) [reply examples/amp-access-fewcents.html:2]: Updated.

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js (hunk 1 above)
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/fewcents-impl.js:149]: Fixed.

**Comment by mayank-s-dev** [comment]
@powerivq The changes which you asked have been fixed.

File: extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js [hunk 3]
```diff
@@ -14,14 +14,34 @@
...
 const TAG = 'amp-access-fewcents';
 
 const TAG_SHORTHAND = 'aaf';
 
+const paywallResponse = {
+  success: true,
+  message: 'Amp article price returned with unlock url.',
+  data: {
+    loginUrl:
+      'https://wallet.hounds.fewcents.co?login_source=amp_paywall&login_publisher_id=47&login_publisher_bid_id=90531&bid_mode=paywall&bid_price=622.38&login_redirect_url=http%3A%2F%2Flocalhost%3A8000%2Fexamples%2Famp-access-fewcents.html&login_publisher_logo_url=https%3A%2F%2Fi1.wp.com%2Fthefix.media%2Fwp-content%2Fuploads%2F2021%2F04%2Flogo-no-margins.png&amp_reader_id=amp-yfpD7ewsxdSdbH4xCboNIw',
```
Comment (powerivq) [inline extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js:28]: Should the URL be `https://wallet.hounds.fewcents.co/?xxxx`? Also on 2 other occasions in this PR. The backslash should be required in URL. Maybe many browsers can work it fine, but still better to follow the standard.

**Review by powerivq — CHANGES_REQUESTED** [review]

File: examples/amp-access-fewcents.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (powerivq) [reply examples/amp-access-fewcents.html:2]: Not pushed yet?

File: extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js [hunk 4]
```diff
@@ -14,14 +14,34 @@
...
+    purchaseOptions: [
+      {
+        price: {
+          currency: 'INR',
+          price: 622.38,
+        },
+        unlockUrl:
+          'https://wallet.hounds.fewcents.co?login_source=amp_paywall&login_publisher_id=47&login_publisher_bid_id=90531&bid_mode=paywall&bid_price=622.38&login_redirect_url=http%3A%2F%2Flocalhost%3A8000%2Fexamples%2Famp-access-fewcents.html&login_publisher_logo_url=https%3A%2F%2Fi1.wp.com%2Fthefix.media%2Fwp-content%2Fuploads%2F2021%2F04%2Flogo-no-margins.png&amp_reader_id=amp-yfpD7ewsxdSdbH4xCboNIw',
```
Comment (powerivq) [inline extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js:36]: Ditto

File: extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js [hunk 5]
```diff
@@ -88,22 +277,30 @@ describes.realWin(
...
           fcButtonText: 'Unlock',
+          fcFewcentsImageRef:
+            'https://dev.fewcents.co/static/media/powered-fewcents.5c8ee304.png',
+          fcTermsRef: 'https://www.fewcents.co/terms',
         };
 
+        vendor.purchaseOptions_ = {
+          price: {currency: 'INR', price: '881'},
+          unlockUrl:
+            'https://wallet.hounds.fewcents.co?login_source=amp…53A8000%252Fexamples%252Famp-access-fewcents.html',
```
Comment (powerivq) [inline extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js:288]: Ditto

**Review by powerivq — CHANGES_REQUESTED** [review]

File: examples/amp-access-fewcents.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (mayank-s-dev) [reply examples/amp-access-fewcents.html]: Fixed and pushed. Earlier I changed for other files.

File: extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js (hunk 3 above)
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js]: Fixed.

File: extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js (hunk 4 above)
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js]: Fixed.

File: extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js (hunk 5 above)
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js]: Fixed.

**Comment by mayank-s-dev** [comment]
@powerivq I have made the changes you asked.

**Review by MichaelRybak — APPROVED** [review]
Approving changes in validator-amp-access-fewcents.html and .out. 

Please wait for approval from @powerivq.

**Comment by mayank-s-dev** [comment]
> Approving changes in validator-amp-access-fewcents.html and .out.
> 
> Please wait for approval from @powerivq.

Thank you @MichaelRybak for your approval.

**Review by powerivq — APPROVED** [review]

**Pull request merged**